### PR TITLE
Extend explanations for some rules

### PR DESCRIPTION
**Pull request by fregante**
- Closes https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1752

### CODE REVIEW

File: docs/rules/explicit-length-check.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
 <!-- end auto-generated rule header -->
 <!-- Do not manually modify this header. Run: `npm run fix:eslint-docs` -->
 
-Enforce explicitly checking the length of an object and enforce the comparison style.
+This rule is only meant to enforce a specific style and make comparisons more clear.
```
Comment (fregante) [inline docs/rules/explicit-length-check.md:10]: The previous description was nearly identical to the title

File: docs/rules/no-array-callback-reference.md [hunk 2]
```diff
@@ -7,6 +7,8 @@
 <!-- end auto-generated rule header -->
 <!-- Do not manually modify this header. Run: `npm run fix:eslint-docs` -->
 
+Passing functions to iterator methods can cause issues when the function is changed without realizing that the iterator passes 2 more parameters to it. **This also applies when using TypeScript,** albeit only if the function accepts the same parameter type used by the iterator method.
```
Comment (fregante) [inline docs/rules/no-array-callback-reference.md:10]: This rule like `no-array-reduce` is very "controversial", so the clearer the better.

File: docs/rules/no-array-reduce.md [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 It's only somewhat useful in the rare case of summing up numbers, which is allowed by default.
 
-Use `eslint-disable` comment if you really need to use it.
+Use `eslint-disable` comment if you really need to use it or disable the rule entirely if you prefer functional programming.
```
Comment (fregante) [inline docs/rules/no-array-reduce.md:12]: You can't change the mind of functional programmers, so it's best to make it clear this is not for them.

File: docs/rules/no-nested-ternary.md [hunk 4]
```diff
@@ -7,7 +7,7 @@
 <!-- end auto-generated rule header -->
 <!-- Do not manually modify this header. Run: `npm run fix:eslint-docs` -->
 
-Improved version of the [`no-nested-ternary`](https://eslint.org/docs/latest/rules/no-nested-ternary) ESLint rule, which allows cases where the nested ternary is only one level and wrapped in parens.
+Improved version of the [`no-nested-ternary`](https://eslint.org/docs/latest/rules/no-nested-ternary) ESLint rule. This rule allows cases where the nested ternary is only one level and wrapped in parens.
```
Comment (fregante) [inline docs/rules/no-nested-ternary.md:10]: "Which" was ambiguous

File: docs/rules/no-null.md [hunk 5]
```diff
@@ -7,7 +7,7 @@
 <!-- end auto-generated rule header -->
 <!-- Do not manually modify this header. Run: `npm run fix:eslint-docs` -->
 
-Disallow the use of the `null` literal, to encourage using `undefined` instead.
+Disallow the use of the `null` literal, to encourage using `undefined` instead. You can find out more about this in: https://github.com/sindresorhus/meta/discussions/7
```
Comment (fregante) [inline docs/rules/no-null.md:10]: Controversial rule. Best to link to the most comprehensive explanation.

File: docs/rules/no-useless-undefined.md [hunk 6]
```diff
@@ -7,6 +7,10 @@
 <!-- end auto-generated rule header -->
 <!-- Do not manually modify this header. Run: `npm run fix:eslint-docs` -->
 
+`undefined` is the default value for new variables, parameters, return statements, etc… so specifying it doesn't make any difference. 
+
+The only case where passing `undefined` is required is due to bad TypeScript types in functions, in which case you can use `checkArguments: false` option.
```
Comment (fregante) [inline docs/rules/no-useless-undefined.md:12]: Is this true? Are there more cases?

File: docs/rules/no-zero-fractions.md [hunk 7]
```diff
@@ -7,7 +7,7 @@
 <!-- end auto-generated rule header -->
 <!-- Do not manually modify this header. Run: `npm run fix:eslint-docs` -->
 
-There is no difference in JavaScript between, for example, `1`, `1.0` and `1.`, so prefer the former for consistency.
+There is no difference in JavaScript between, for example, `1`, `1.0` and `1.`, so prefer the former for consistency and brevity.
```
Comment (fregante) [inline docs/rules/no-zero-fractions.md:10]: What's a word for "completely useless and redundant"? 😅

File: docs/rules/prefer-module.md [hunk 8]
```diff
@@ -7,7 +7,7 @@
 <!-- end auto-generated rule header -->
 <!-- Do not manually modify this header. Run: `npm run fix:eslint-docs` -->
 
-Prefer using the [JavaScript module](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) format over the legacy CommonJS module format.
+Prefer using the [JavaScript module](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) format over the legacy CommonJS module format. Together with other changes, this helps the ecosystem migrate to a single, native module format.
```
Comment (fregante) [inline docs/rules/prefer-module.md:10]: This is "controversial" and difficult rule to fix. I think we should add even more than this reasoning without starting CJS vs ESM flames.

File: docs/rules/prefer-number-properties.md [hunk 9]
```diff
@@ -7,7 +7,7 @@
 <!-- end auto-generated rule header -->
 <!-- Do not manually modify this header. Run: `npm run fix:eslint-docs` -->
 
-Enforces the use of:
+ECMAScript 2015 moved globals onto the `Number` constructor for consistency and to slightly improve them. This rule enforces their usage to limit the usage of globals:
```
Comment (fregante) [inline docs/rules/prefer-number-properties.md:10]: I couldn't find why they did this, this is my assumption.

File: docs/rules/prefer-query-selector.md [hunk 10]
```diff
@@ -7,7 +7,7 @@
 <!-- end auto-generated rule header -->
 <!-- Do not manually modify this header. Run: `npm run fix:eslint-docs` -->
 
-It's better to use the same method to query DOM elements.
+It's better to use the same method to query DOM elements. This helps keep consistency and it lends itself to future improvements (e.g. more specific selectors).
```
Comment (fregante) [inline docs/rules/prefer-query-selector.md:10]: Best I could do. The rule is more about style than utility (`getElementsByTagName` looks obsolete).

File: docs/rules/no-zero-fractions.md (hunk 7 above)
Comment (fisker) [reply docs/rules/no-zero-fractions.md:10]: `1..toString()`is not.useless.

File: docs/rules/prefer-prototype-methods.md [hunk 11]
```diff
@@ -31,6 +31,7 @@ const array = Array.prototype.slice.apply(bar);
 
 ```js
 const hasProperty = Object.prototype.hasOwnProperty.call(foo, 'property');
+// Or better yet Object.hasOwn(foo, 'property')
```
Comment (fisker) [inline docs/rules/prefer-prototype-methods.md:34]: We can use `Object.prototype.toString` as example.

File: docs/rules/no-zero-fractions.md (hunk 7 above)
Comment (fregante) [reply docs/rules/no-zero-fractions.md:10]: The rule doesn't affect that, so why are you mentioning it?

Comment (fisker) [reply docs/rules/no-zero-fractions.md:10]: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/f14433068bbf92d56b3aaa457c5a1aa0ec19e7ab/test/snapshots/no-zero-fractions.mjs.md?plain=1#L2630

**Pull request merged**